crates/rover-client/src/query/graph/introspect.rs

```diff
@@ -0,0 +1,130 @@
+use std::{collections::HashMap, convert::TryFrom};
+
+use crate::blocking::Client;
+use crate::introspection::Schema;
+use crate::RoverClientError;
+use graphql_client::*;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    query_path = "src/query/graph/introspect_query.graphql",
+    schema_path = "src/query/graph/introspect_schema.graphql",
+    response_derives = "PartialEq, Debug, Serialize, Deserialize",
+    deprecated = "warn"
+)]
+
+/// This struct is used to generate the module containing `Variables` and
+/// `ResponseData` structs.
+/// Snake case of this name is the mod name. i.e. introspection_query
+pub struct IntrospectionQuery;
+
+/// this struct contains all the info needed to print the result of the delete.
+/// `updated_gateway` is true when composition succeeds and the gateway config
+/// is updated for the gateway to consume. `composition_errors` is just a list
+/// of strings for when there are composition errors as a result of the delete.
+#[derive(Debug, PartialEq)]
+pub struct IntrospectionResponse {
+    pub result: String,
+}
```

Comment:
Not sure I understand where this comes form or where it is used? Why is this about the gateway config?

Reply:
the comment is old, sorry about that. 

`IntrospectionResponse` is used in this module as the response type we sent throughout to Rover. It's ends be being the encoded schema result we get back from the encoder.